Quadriga/src/main/java/edu/asu/spring/quadriga/web/config/SimpleSignInAdapter.java

```diff
@@ -0,0 +1,123 @@
+
+    private IUserManager userManager;
+    private IUserHelper userHelper;
+
+    public SimpleSignInAdapter(IUserManager userManager, IUserHelper userHelper) {
+        this.userManager = userManager;
+        this.userHelper = userHelper;
+    }
+
+    /**
+     * This method checks if Quadriga contains details of the user authenticated
+     * by Github. If the user details are present (which indicates the user is
+     * an authorized user), appropriate roles are assigned, Else if the user
+     * details are not present (which indicates the user is an unregistered /
+     * not approved user), restricted access is assigned.
+     */
+
+    @Override
+    public String signIn(String userId, Connection<?> connection, NativeWebRequest request) {
+
+        List<GrantedAuthority> authorities = new ArrayList<>();
+        SocialSignInStatus messageType = null;
+        IUser user = (IUser) userManager.findUserByProviderUserId(connection.getKey().getProviderUserId(),
+                connection.getKey().getProviderId());
+        // if user details is not present in the database, create a database
+        // record for the user.  
+        if (user == null) {
+            
+            try {
+                    user = userHelper.createUser(userId, connection);
+                    AccountRequest accountRequest = new AccountRequest();
+                    accountRequest.setUsername(user.getUserName());
+                    accountRequest.setName(user.getName());
+                    accountRequest.setEmail(user.getEmail());
+                    accountRequest.setSocialSignIn(true);
+                    accountRequest.setProvider(user.getProvider());
+                    accountRequest.setUserIdOfProvider(user.getUserIdOfProvider());
+                    userManager.addNewUser(accountRequest);
+                    messageType = SocialSignInStatus.REGISTRATION_SUCCESS;
+            } catch (QuadrigaStorageException e){
+                logger.error("Could not add user.", e);
+                messageType = SocialSignInStatus.REGISTRATION_FAILED;
+            } catch(UsernameExistsException e){ 
+                logger.error("User request exists.", e);
+                messageType = SocialSignInStatus.ACCOUNT_APPROVAL_PENDING;
+            }catch (QuadrigaNotificationException e) {
+                logger.error("Could not notify admin about the new user.", e);
+            } 
+        }
+        // if user details is present in the database, assign appropriate roles
+        // to the user.
+        else {
+
+            List<IQuadrigaRole> quadrigaRoles = user.getQuadrigaRoles();
+            if (quadrigaRoles != null) {
+                for (IQuadrigaRole quadrigaRole : quadrigaRoles) {
+                    authorities.add(new QuadrigaGrantedAuthority(quadrigaRole.getId()));
+                }
+            }
+        }
+
+        QuadrigaUserDetails userDetails = null;
```

Comment:
move after if statement (it's not used in the if statement, so it should be before it)

Reply:
actually just merge with line 108